lib/cloud/imds/gcp/imds.go

```diff
@@ -53,24 +52,75 @@ func (rt contextRoundTripper) RoundTrip(req *http.Request) (*http.Response, erro
 
 type metadataGetter func(ctx context.Context, path string) (string, error)
 
+// InstanceRequest contains parameters for making a request to a specific instance.
+type InstanceRequest struct {
+	// ProjectID is the ID of the VM's project.
+	ProjectID string
+	// Zone is the instance's zone.
+	Zone string
+	// Name is the instance's name.
+	Name string
+	// ID is the instance's ID.
+	ID uint64
+	// WithoutHostKeys indicates that the client should not request the instance's
+	// host keys.
+	WithoutHostKeys bool
+}
+
+func (req *InstanceRequest) CheckAndSetDefaults() error {
+	if req.ProjectID == "" {
+		return trace.BadParameter("projectID must be set")
+	}
+	if req.Zone == "" {
+		return trace.BadParameter("zone must be set")
+	}
+	if req.Name == "" {
+		return trace.BadParameter("name must be set")
+	}
+	return nil
+}
+
+// Instance represents a GCP VM.
+type Instance struct {
+	// Name is the instance's name.
+	Name string
+	// Zone is the instance's zone.
+	Zone string
+	// ProjectID is the ID of the project the VM is in.
+	ProjectID string
+	// ServiceAccount is the email address of the VM's service account, if any.
+	ServiceAccount string
+	// Labels is the instance's labels.
+	Labels map[string]string
+
+	InternalIPAddress string
+	ExternalIPAddress string
+	HostKeys          []ssh.PublicKey
+	FingerPrint       string
+	Kind              string
+	Items             map[string]string
```

Comment:
```suggestion
	MetadataItems     map[string]string
```
or something like that, so it's clearer what "items" we're talking about.